### PR TITLE
Storage compatibility for HNSW graph file

### PR DESCRIPTION
**Pull request by agourlay**
This PR adds the HNSW's `graph.bin` to the storage compatibility test (https://github.com/qdrant/qdrant/issues/867)

I have decided to extract an independent script to generate the data for the storage archive. 

It now injects 100 vectors with dim. 50 using a payload index.

I have validated that the new archive contains the expected files.

```
+ tar -xvjf ./tests/storage-compat/storage.tar.bz2
storage/
storage/raft_state
storage/collections/
storage/collections/test_collection/
storage/collections/test_collection/version.info
storage/collections/test_collection/0/
storage/collections/test_collection/0/temp_segments/
storage/collections/test_collection/0/segments/
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/000016.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/IDENTITY
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/LOCK
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/vector_index/
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/vector_index/hnsw_config.json
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/vector_index/graph.bin
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/000017.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/000024.log
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/version.info
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/000018.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/000016.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/IDENTITY
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/OPTIONS-000013
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/LOG.old.1659105084317709
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/000014.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/LOCK
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/000015.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/000021.log
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/MANIFEST-000020
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/LOG
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/OPTIONS-000023
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/config.json
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/payload_index/CURRENT
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/MANIFEST-000023
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/OPTIONS-000026
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/LOG.old.1659105084237516
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/LOG
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/000019.sst
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/OPTIONS-000015
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/segment.json
storage/collections/test_collection/0/segments/f1234b3e-ca06-479e-92bb-d7e3416c84fe/CURRENT
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/000005.log
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/IDENTITY
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/OPTIONS-000013
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/LOCK
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/version.info
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/000005.log
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/IDENTITY
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/OPTIONS-000013
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/LOCK
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/LOG
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/MANIFEST-000004
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/OPTIONS-000011
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/config.json
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/payload_index/CURRENT
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/LOG
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/MANIFEST-000004
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/OPTIONS-000015
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/segment.json
storage/collections/test_collection/0/segments/9e702d43-5a14-4622-b8e6-357dc365f9f5/CURRENT
storage/collections/test_collection/0/shard_config.json
storage/collections/test_collection/0/wal/
storage/collections/test_collection/0/wal/open-2
storage/collections/test_collection/0/wal/open-1
storage/collections/test_collection/config.json
storage/collections_meta_wal/
storage/collections_meta_wal/open-2
storage/collections_meta_wal/open-1
storage/aliases/
storage/aliases/data.json
```

I took the opportunity to improve the documentation regarding refreshing the storage archive to spread the knowledge.

### CODE REVIEW

**Comment by generall** [comment]
That is strange, that `gen_storage_compat_data.sh` created a vector index, there is no explicit config there to force optimizer to do so. Are you sure you didn't change `config.yaml` or `production.yaml`? I also tried to run the script locally and it didn't create vector index for me

File: tests/storage-compat/gen_storage_compat_data.sh [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+# create collection
+curl -X PUT "http://$QDRANT_HOST/collections/test_collection" \
+  -H 'Content-Type: application/json' \
+  --fail -s \
+  --data-raw '{
+      "vector_size": 50,
+      "distance": "Dot",
+      "optimizers_config": {
+        "default_segment_number": 2
```
Comment (generall) [inline tests/storage-compat/gen_storage_compat_data.sh:21]:
```suggestion
        "default_segment_number": 2,
        "indexing_threshold_kb": 10
```

**Comment by agourlay** [comment]
> there is no explicit config there to force optimizer to do so. Are you sure you didn't change config.yaml or production.yaml?

I did change the configuration manually, it is explained in the new documentation :)

Would you prefer the script to perform the config. change directly?

File: tests/storage-compat/gen_storage_compat_data.sh (hunk 1 above)
Comment (agourlay) [reply tests/storage-compat/gen_storage_compat_data.sh:21]: Just understood this comment :+1: 
I will apply this change and update the docs to remove the manual config change step.

Comment (agourlay) [reply tests/storage-compat/gen_storage_compat_data.sh]: https://github.com/qdrant/qdrant/pull/875/commits/ccbedfe71a736772242f04fc03be81327156726b

**Review by generall — APPROVED** [review]

**Pull request merged**